### PR TITLE
OVN-Kubernetes: clarify test plan, latency commitments

### PR DESCRIPTION
**Pull request by squeed**
@smarterclayton asked some good questions in #30 after it was merged. This starts to answer that.

### CODE REVIEW

File: enhancements/20190919-OVN-Kubernetes.md [hunk 1]
```diff
@@ -48,39 +48,50 @@ ovn-northd, which converts logical objects in to flows.
...
 
-The node-level process (ovn-southd) is responsible for programming the
+The node-level process (ovn-controller) is responsible for programming the
 flows from ovn-northd in to the vswitch on the local node. It also handles
 some bookkeeping.
 
 #### Controller
 There will be a controller daemonset on every master node. They will
 perform leader election directly via the Kubernetes API. The ovn-kubernetes
 process will, when it achieves leadership, will be responsible for
```
Comment (danwinship) [inline enhancements/20190919-OVN-Kubernetes.md]: "will" twice

File: enhancements/20190919-OVN-Kubernetes.md [hunk 2]
```diff
@@ -48,39 +48,50 @@ ovn-northd, which converts logical objects in to flows.
...
 #### Node-level
 The node-level components are configured as a daemonset. They run on all
 nodes, including the control-plane. They watch for endpoint changes,
 watch the sbdb for changes, and configure the nodes.
 
 ### Test Plan
-OVN-Kubernetes must pass all standard QE tests, as well as the full e2e
-suite. It does not need any special testing, as the GA release does not
-enable any new features.
+In addition to the existing CI and QE test suites, we will develop both manual and
```
Comment (danwinship) [inline enhancements/20190919-OVN-Kubernetes.md]: aside: we need e2e-aws-upgrade-ovn-kubernetes (*after* we implement the HA stuff)

File: enhancements/20190919-OVN-Kubernetes.md [hunk 3]
```diff
@@ -48,39 +48,50 @@ ovn-northd, which converts logical objects in to flows.
...
 
 ### Graduation Criteria
 GA:
 - OVN-Kubernetes is stable
-- The control-plane is low-latency: changes are installed on the nodes in under 10 seconds
-- Reliable control plane: losing a master node does not cause more than a few seconds of extra latency, and no loss of network connectivity
-- Reliable node: Upgrading ovn-kubernetes components
+- The control-plane is low-latency. For small clusters (sub 1000 pods),
+    - service changes: 95%ile 5 seconds, 99%ile 15 seconds
+    - pod creation (time blocked on networking): 95%ile 15 seconds, 99%ile 30 seconds
```
Comment (danwinship) [inline enhancements/20190919-OVN-Kubernetes.md]: NetworkPolicy changes

File: enhancements/20190919-OVN-Kubernetes.md [hunk 4]
```diff
@@ -48,39 +48,50 @@ ovn-northd, which converts logical objects in to flows.
...
+- Reliable control plane: losing a master node does not cause more than 10 seconds of
```
Comment (danwinship) [inline enhancements/20190919-OVN-Kubernetes.md]: That is also assuming the "small cluster" ?

File: enhancements/20190919-OVN-Kubernetes.md [hunk 5]
```diff
@@ -107,6 +118,12 @@ version skew can occur between nodes.
 
 ## Drawbacks
 
+OVN currently has no way to implement:
+- Session affinity for Services
+- "except" blocks for NetworkPolicy
```
Comment (danwinship) [inline enhancements/20190919-OVN-Kubernetes.md]: Neither does openshift-sdn...

File: enhancements/20190919-OVN-Kubernetes.md (hunk 3 above)
Comment (squeed) [reply enhancements/20190919-OVN-Kubernetes.md]: Do you have any suggestions for numbers?

File: enhancements/20190919-OVN-Kubernetes.md (hunk 4 above)
Comment (squeed) [reply enhancements/20190919-OVN-Kubernetes.md]: Ideally, no.

File: enhancements/20190919-OVN-Kubernetes.md [hunk 6]
```diff
@@ -48,39 +48,50 @@ ovn-northd, which converts logical objects in to flows.
 The data store is a replicated database, with a traditional master-backup
 architecture.
 
-The node-level process (ovn-southd) is responsible for programming the
+The node-level process (ovn-controller) is responsible for programming the
```
Comment (dcbw) [inline enhancements/20190919-OVN-Kubernetes.md]: Actually this should be "is responsible for programming flows from the OVN southbound database into the vswitch on the local node".

File: enhancements/20190919-OVN-Kubernetes.md (hunk 3 above)
Comment (dcbw) [inline enhancements/20190919-OVN-Kubernetes.md]: @squeed these are pretty generous latencies for pod creation :) We were originally targeting 95%ile = 2 seconds at the OVN level itself, and maybe the whole CRIO -> CNI -> etc adds a tiny bit, but not much. So if anyone objects to these numbers, they could come down somewhat.

File: enhancements/20190919-OVN-Kubernetes.md (hunk 1 above)
Comment (russellb) [reply enhancements/20190919-OVN-Kubernetes.md]: this was fixed

File: enhancements/20190919-OVN-Kubernetes.md (hunk 3 above)
Comment (russellb) [reply enhancements/20190919-OVN-Kubernetes.md]: networkpolicy numbers were added

File: enhancements/20190919-OVN-Kubernetes.md (hunk 5 above)
Comment (russellb) [reply enhancements/20190919-OVN-Kubernetes.md]: It sounds like adding this clarification would be helpful

File: enhancements/20190919-OVN-Kubernetes.md (hunk 6 above)
Comment (russellb) [reply enhancements/20190919-OVN-Kubernetes.md]: this change still needed

**Comment by russellb** [comment]
@squeed Can you take a look at this stale enhancement?  I see a couple of minor changes from past review that could be quickly applied.  Then I think we should just merge this.

**Pull request closed without merging**